builder/src/core/node.rs

```diff
@@ -388,15 +394,18 @@ impl Node {
     }
 
     /// Dump a chunk from u8 slice into the data blob.
-    /// Return `BlobChunkInfoV2Ondisk` when the chunk is added into a batch chunk.
+    /// Return `BlobChunkInfoV2Ondisk` iff the chunk is added into a batch chunk.
+    /// Return dumped size iff not `BlobFeatures::SEPARATE`.
+    /// Dumped size can be zero if chunk data is cached in Batch Generator,
+    /// and may contain previous chunk data cached in Batch Generator.
     fn dump_file_chunk(
         &self,
         ctx: &BuildContext,
         blob_ctx: &mut BlobContext,
         blob_writer: &mut ArtifactWriter,
         chunk_data: &[u8],
         chunk: &mut ChunkWrapper,
-    ) -> Result<Option<BlobChunkInfoV2Ondisk>> {
+    ) -> Result<(Option<BlobChunkInfoV2Ondisk>, Option<u32>)> {
```

Comment:
How about returning `Result<(Option<BlobChunkInfoV2Ondisk>, u32)>`, which may help to simplify the implementation?

Reply:
fixed.